core/sqf/src/seatrans/tm/hbasetmlib2/src/main/java/org/trafodion/dtm/HBaseTxClient.java

```diff
@@ -1033,17 +961,16 @@ public void run() {
                         Map<String, byte[]> regions = null;
                         Map<Long, TransactionState> transactionStates =
                                 new HashMap<Long, TransactionState>();
-                        try {
-                            regions = zookeeper.checkForRecovery();
-                        } catch (Exception e) {
-                            if (regions != null) { // ignore no object returned by zookeeper.checkForRecovery
-                               LOG.error("An ERROR occurred while checking for regions to recover. " + "TM: " + tmID);
-                               StringWriter sw = new StringWriter();
-                               PrintWriter pw = new PrintWriter(sw);
-                               e.printStackTrace(pw);
-                               LOG.error(sw.toString());
-                            }
-                        }
+                        boolean loopBack = false;
+                        do {
+                           try {
+                               regions = zookeeper.checkForRecovery();
```

Comment:
Should there be a loopBack = false after checkForRecovery in the case where an exception occurred but then the operation succeeded?